.gitignore

```diff
@@ -24,6 +24,7 @@ cabal.project.local~
 .ghc.environment.*
 .ghci
 *.tix
+*.lock
```

Comment:
Perhaps `stack.yaml.lock` should be checked in to git, as package lock files usually are.